### PR TITLE
fix(label-content-name-mismatch): ignore non `widget` aria role(s) & do not use deprecated `lookupTable.rolesOfType`

### PR DESCRIPTION
**Pull request by jkodu**
Updated lookUpTable with roles of type `widget`.

Closes issue:
- https://github.com/dequelabs/axe-core/issues/2019

## Reviewer checks

**Required fields, to be filled out by PR reviewer(s)**

- [x] Follows the commit message policy, appropriate for next version
- [x] Code is reviewed for security

### CODE REVIEW

**Comment by straker** [comment]
Thanks for posting the new sizes. With axe-core 4.0 being the next phase, I want to introduce a size budget (#1930). Adding 100 KB of / 60 KB  (min) to axe-core seems a bit much just to fix a hard-coded list we have already.

I know we've talked about aria-query before and using it in place of our lookup table entirely. @WilcoFiers  is that something we could do for 4.0?

File: lib/core/imports/index.js [hunk 1]
```diff
@@ -47,5 +47,6 @@ axe.imports = {
 	CssSelectorParser: require('css-selector-parser').CssSelectorParser,
 	doT: require('@deque/dot'),
 	emojiRegexText: require('emoji-regex'),
-	memoize: require('memoizee')
+	memoize: require('memoizee'),
+	ariaQuery: require('aria-query')
```
Comment (WilcoFiers) [inline lib/core/imports/index.js]: I don't think we should pull aria-query into axe-core at this point. There's a much simpler fix for this. Please don't overthink it.

**Comment by stephenmathieson** [comment]
I agree with @WilcoFiers and @straker. Adding 100k just to fix a single bug probably isn't worth it.

File: lib/commons/aria/index.js [hunk 2]
```diff
@@ -2387,43 +2387,54 @@ lookupTable.evaluateRoleForElement = {
...
  * 	"alert", "alertdialog", "button", "checkbox", "dialog", "gridcell", "link", "log", "marquee", "menuitem", "menuitemcheckbox",
  * 	"menuitemradio", "option", "progressbar", "radio", "scrollbar", "searchbox", "slider", "spinbutton", "status", "switch", "tab", "tabpanel",
  * 	"textbox", "timer", "tooltip", "treeitem"
  * ]
- * There are some differences against specs, hence the below listing was made
+ * There are some differences against specs, hence the below listing was made by cross referring with `aria-query`
  */
 lookupTable.rolesOfType = {
 	widget: [
+		'command',
```
Comment (straker) [inline lib/commons/aria/index.js]: command is an abstract role, not a widget role. Same for a lot of the ones you've added. Axe doesn't allow abstract roles on elements so we don't need to add them here.

Also don't we need to change `tooltip` in the a `lookupTable.role` to not be a widget role? I'm not sure how we differentiate between `widget` roles and `rolesOfType.widget` (such as `alertdialog` being a widget in the lookupTable but not in this list)

Comment (jkodu) [reply lib/commons/aria/index.js]: Thanks @straker. 

The reason I did not use the above code snippet is our lookupTable is out of date & a small change to accommodate something like this usually breaks other tests. 

We have had conversations to update the lookupTable in the past. I used aria-query to generate this list of roles with a widget role type.

Let me discuss with @WilcoFiers for the right way forward to be lookupTable.

**Comment by jkodu** [comment]
Ended up removing the array `rolesOfType.widget` and changed the type of a role in the lookup table.

History points to this [snapshot](https://github.com/dequelabs/axe-core/commit/a4255da17c736bfa740790e61f8f69635866d4da#diff-ec67bb6113bfd9a900ee27ecef942f74R2372) when this listing was added. From memory I believe there was an ask to not touch/ alter the lookup table when this PR was authored.

**Comment by straker** [comment]
@WilcoFiers probably needs to approve removing the list as he'll have the most knowledge of why it was created in the first place.

File: lib/commons/aria/index.js [hunk 3]
```diff
@@ -2385,45 +2385,3 @@ lookupTable.evaluateRoleForElement = {
...
- * The current lookupTable.role['widget'] widget, yeilds
- * ->
- * [
- * 	"alert", "alertdialog", "button", "checkbox", "dialog", "gridcell", "link", "log", "marquee", "menuitem", "menuitemcheckbox",
- * 	"menuitemradio", "option", "progressbar", "radio", "scrollbar", "searchbox", "slider", "spinbutton", "status", "switch", "tab", "tabpanel",
- * 	"textbox", "timer", "tooltip", "treeitem"
- * ]
- * There are some differences against specs, hence the below listing was made
- */
-lookupTable.rolesOfType = {
```
Comment (WilcoFiers) [inline lib/commons/aria/index.js]: Taking this out could break custom rules. Please leave it in, label it as deprecated, and make sure to include the word "deprecated" in the title of the PR so it appears in the changelog.

**Pull request merged**